resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/Comparators.java

```diff
@@ -327,15 +330,35 @@ public int compare(ResourceAddress addr1, ResourceAddress addr2) {
 
             ResourceAddress transport1 = addr1.getOption(TRANSPORT);
             ResourceAddress transport2 = addr2.getOption(TRANSPORT);
-            
+
             int compareTransport = PROTOCOL_STACK_COMPARATOR.compare(transport1, transport2);
             if (compareTransport != 0) {
                 return compareTransport;
             }
-            
+
+            int compareExternalURI = EXTERNAL_URI_COMPARATOR.compare(addr1, addr2);
+            if (compareExternalURI == 0) {
```

Comment:
We shouldn't use external URI in comparisons (for e.g dns might map foo.com and bar.com to the same ip).